sdk/appconfiguration/azure-data-appconfiguration/src/main/java/com/azure/data/appconfiguration/implementation/SyncTokenPolicy.java

```diff
@@ -0,0 +1,74 @@
+// Copyright (c) Microsoft Corporation. All rights reserved.
+// Licensed under the MIT License.
+
+package com.azure.data.appconfiguration.implementation;
+
+import com.azure.core.http.HttpPipelineCallContext;
+import com.azure.core.http.HttpPipelineNextPolicy;
+import com.azure.core.http.HttpResponse;
+import com.azure.core.http.policy.HttpPipelinePolicy;
+import reactor.core.publisher.Mono;
+
+import java.util.concurrent.ConcurrentHashMap;
+import java.util.stream.Collectors;
+
+public final class SyncTokenPolicy implements HttpPipelinePolicy {
+    private static final String SYNC_TOKEN = "Sync-Token";
+    private final ConcurrentHashMap<String, SyncToken> syncTokenMap; // key is sync-token id
+
+    public SyncTokenPolicy() {
+        syncTokenMap = new ConcurrentHashMap<>();
+    }
+
+    /**
+     * Add or update the sync token id and value to a thread safe map.
+     *
+     * @param context request context
+     * @param next The next policy to invoke.
+     * @return
+     */
+    @Override
+    public Mono<HttpResponse> process(HttpPipelineCallContext context, HttpPipelineNextPolicy next) {
+
+        // attempt to add header to the request
```

Comment:
Attempting? Are there scenarios where the header value is not added?

Reply:
updated